TELCODOCS-2036: Procedure added for MLX secure boot (#85440)
openshift/openshift-docs: modules/nw-sriov-nic-mlx-secure-boot.adoc

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

// :_mod-docs-content-type: PROCEDURE
[id="nw-sriov-nic-mlx-secure-boot_{context}"]

> = Mellanox Secure Boot

Reviewer: maybe:  Configure SR-IOV on Mellanox cards when Secure Boot is enabled

Author: Thanks @SchSeba for your feedback. I have renamed the title 'Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled'

> Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV Network Operator. This allows {product-title} customers to use the existing virtual functions in the Mellanox cards firmware, and is currently the only way to use Mellanox NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Reviewer: I think here it should be something like.

The SR-IOV operator supports an option to skip the firmware configuration for Mellanox devices. That is currently the only way to be able and create virtual functions via the SR-IOV Network operator if the system has secure boot enabled.

This will require a manual configuration from the user to allocate the number of virtual functions in the firmware before switching the system to secure boot.

Note: the number of virtual functions in the firmware will be the max number of virtual functions the user can request in the policy (meaning the user can create less)

Author: Thanks @SchSeba for your feedback. I have rewritten the opening paragraph and added the Note.

.Procedure

. Configure the virtual functions (VFs).

.. Run the following command when the system is without a secure boot when using the sriov-config daemon:
+
[source,terminal]
----
mstconfig -d -0001:b1:00.0 set SRIOV_EN=1 NUM_OF_VFS=16
----

.. Reboot the system to enable the virtual functions.

> . Configure the SR-IOV Network Operator with the Mellanox plugin disabled.

Reviewer: maybe switching the section to configure secure boot after this change (and reboot the system) this will reduce the number if required reboots by one

Author: Thanks @SchSeba for your feedback. I have applied your suggestion.

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

[id="nw-sriov-nic-mlx-secure-boot_{context}"]
= Mellanox Secure Boot

> Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Author: Updated with the ocpdocs-vale-bot recommendation.

> Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Author: Updated with the ocpdocs-vale-bot recommendation.

> Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Author: Thanks @cgoncalves for your feedback. I have applied your suggestions.

.Procedure

. Configure the virtual functions (VFs)

.. Run the following command when the system is without a secure boot when using the sriov-config daemon:
+
[source,terminal]
----
mstconfig -d -0001:b1:00.0 set SRIOV_EN=1 NUM_OF_VFS=16
----

.. Reboot the system to enable the virtual functions.

> . Configure the sriov operator with the Mellanox plugin disabled

Reviewer 2: ```suggestion
. Configure the SR-IOV Network Operator with the Mellanox plugin disabled
```

Author: Thanks @cgoncalves for your feedback. I have applied your suggestion.

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

// :_mod-docs-content-type: PROCEDURE
[id="nw-sriov-nic-mlx-secure-boot_{context}"]
= Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled

The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox (MLX) devices. It is currently the only way to create virtual functions by using the SR-IOV Network operator, if the system has secure boot enabled. You are required to manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.

[NOTE]
====
The number of virtual functions in the firmware will be the maximum number of virtual functions that you can request in the policy.
====

.Procedure

. Configure the virtual functions (VFs).

.. Run the following command when the system is without a secure boot when using the sriov-config daemon:
+
[source,terminal]
----

> mstconfig -d -0001:b1:00.0 set SRIOV_EN=1 NUM_OF_VFS=16

Reviewer: `-0001:b1:00.0` -> `0001:b1:00.0`

also can you put a small note about the parameters?

1. <PF-pci-address> 
2. enable sriov support on the card
3. number of virtual functions to enable in the card firmware

----

. Configure the SR-IOV Network Operator with the Mellanox plugin disabled.
+
[source,yaml]
----
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovOperatorConfig
metadata:
  name: default
  namespace: openshift-sriov-network-operator
spec:
  configDaemonNodeSelector: {}
  configurationMode: daemon
  disableDrain: false
  disablePlugins:
  - mellanox
  enableInjector: true
  enableOperatorWebhook: true
  logLevel: 2
----

. Reboot the system to enable the virtual functions and the configuration settings.

. Check the virtual functions (VFs) after rebooting.
+
[source,terminal]
----

> # oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io  cnfdr14.telco5g.eng.rdu2.redhat.com -oyaml

Reviewer: lets not expose our server names `cnfdr14.telco5g.eng.rdu2.redhat.com` -> `worker-0`

Author: Thanks @SchSeba for your feedback. I have applied your suggestion.

----
+
[source,yaml]
----
- deviceID: 101d
    driver: mlx5_core
    eSwitchMode: legacy
    linkSpeed: -1 Mb/s
    linkType: ETH
    mac: 08:c0:eb:96:31:25
    mtu: 1500
    name: ens3f1np1
    pciAddress: 0000:b1:00.1

>     totalvfs: 16

Reviewer: please add a note to point this is the number that was configured in the mstconfig for the firmware

Reviewer: @ogradyp not the `pciAddress` it's the `totalvfs` it should be the same number that the customer apply in `NUM_OF_VFS=16` 

command from above

`mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>`

Author: Thanks @SchSeba for your feedback. I have applied your suggestion.

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-sriov-nic-mlx-secure-boot_{context}"]
= Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled

> The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox (MLX) devices. It is currently the only way to create virtual functions by using the SR-IOV Network Operator, if the system has secure boot enabled. You are required to manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.

Reviewer 3: I don't see "MLX" in any official docs or web presence for this brand
```suggestion
The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox devices. This option allows you to create virtual functions by using the SR-IOV Network Operator when the system has secure boot enabled. You must manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.
```

Author: Thanks @jeana-redhat for your feedback. I have applied your suggestion.

[NOTE]
====

> The number of virtual functions in the firmware are the maximum number of virtual functions that you can request in the policy.

Reviewer 3: ```suggestion
The number of virtual functions in the firmware is the maximum number of virtual functions that you can request in the policy.
```

Author: Thanks @jeana-redhat for your feedback. I have applied your suggestion.

====

.Procedure

. Configure the virtual functions (VFs):

> .. Run the following command when the system is without a secure boot when using the sriov-config daemon:

Reviewer 3: this step has a single substep - this should just be a step.

Author: Thanks @jeana-redhat for your feedback. I have applied your suggestion.

+
[source,terminal]
----
$ mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>
----
<1> The `SRIOV_EN` environment variable enables the SR-IOV Network Operator support on the Mellanox card.
<2> The `NUM_OF_VFS` environment variable specifies the number of virtual functions to enable in the firmware.

. Configure the SR-IOV Network Operator by disabling the Mellanox plugin. See the following `SriovOperatorConfig` example configuration:
+
[source,yaml]
----
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovOperatorConfig
metadata:
  name: default
  namespace: openshift-sriov-network-operator
spec:
  configDaemonNodeSelector: {}
  configurationMode: daemon
  disableDrain: false
  disablePlugins:
  - mellanox
  enableInjector: true
  enableOperatorWebhook: true
  logLevel: 2
----

. Reboot the system to enable the virtual functions and the configuration settings.

. Check the virtual functions (VFs) after rebooting the system by running the following command:
+
[source,terminal]
----
$ oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io worker-0 -oyaml
----
+

> [source,yaml]

Reviewer 3: Is this sample output?
```suggestion
.Example output
[source,yaml]
```

Author: Thanks @jeana-redhat for your feedback. Yes. I have applied your suggestion.

----
- deviceID: 101d
    driver: mlx5_core
    eSwitchMode: legacy
    linkSpeed: -1 Mb/s
    linkType: ETH
    mac: 08:c0:eb:96:31:25
    mtu: 1500
    name: ens3f1np1
    pciAddress: 0000:b1:00.1 <1>
    totalvfs: 16
    vendor: 15b3
----
<1> The `totalvfs` value is the same number used in the `mstconfig` command earlier in the procedure. 

> . Enable secure boot to improve your system's security by preventing unauthorized operating systems and malicious software from loading during the device's boot process. 

Reviewer 3: Not sure if we want to make this claim
```suggestion
. Enable secure boot to prevent unauthorized operating systems and malicious software from loading during the device's boot process. 
```

Author: Thanks @jeana-redhat for your feedback. I have applied your suggestion.

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

> // :_mod-docs-content-type: PROCEDURE

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

[id="nw-sriov-nic-mlx-secure-boot_{context}"]
= Configuring the SR-IOV Network Operator on Mellanox cards when Secure Boot is enabled

> The SR-IOV Network Operator supports an option to skip the firmware configuration for Mellanox (MLX) devices. It is currently the only way to create virtual functions by using the SR-IOV Network operator, if the system has secure boot enabled. You are required to manually configure and allocate the number of virtual functions in the firmware before switching the system to secure boot.

Author: Thanks @xenolinux for your feedback. All relevant instances of Operator and now capitalized.

[NOTE]
====

> The number of virtual functions in the firmware will be the maximum number of virtual functions that you can request in the policy.

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

====

.Procedure

> . Configure the virtual functions (VFs).

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

.. Run the following command when the system is without a secure boot when using the sriov-config daemon:
+
[source,terminal]
----

> mstconfig -d -0001:b1:00.1 set SRIOV_EN=1 NUM_OF_VFS=16 <1> <2>

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

----
<1> `SRIOV_EN` Enable SR-IOV Network operator support on the Mellanox card.

> <2> `NUM_OF_VFS` Specify the number of virtual functions to enable in the firmware.

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

> . Configure the SR-IOV Network Operator with the Mellanox plugin disabled.

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

+
[source,yaml]
----
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovOperatorConfig
metadata:
  name: default
  namespace: openshift-sriov-network-operator
spec:
  configDaemonNodeSelector: {}
  configurationMode: daemon
  disableDrain: false
  disablePlugins:
  - mellanox
  enableInjector: true
  enableOperatorWebhook: true
  logLevel: 2
----

> . Reboot the system to enable the virtual functions and the configuration settings.

Author: Thanks @xenolinux for your feedback. Yes reboot is suitable here.

. Check the virtual functions (VFs) after rebooting.
+
[source,terminal]
----

> # oc -n openshift-sriov-network-operator get sriovnetworknodestate.sriovnetwork.openshift.io worker-0 -oyaml

Author: Thanks @xenolinux for your feedback. I have applied your suggestion.

----
+
[source,yaml]
----
- deviceID: 101d
    driver: mlx5_core
    eSwitchMode: legacy
    linkSpeed: -1 Mb/s
    linkType: ETH
    mac: 08:c0:eb:96:31:25
    mtu: 1500
    name: ens3f1np1
    pciAddress: 0000:b1:00.1 <1>
    totalvfs: 16
    vendor: 15b3
----
<1> The `totalvfs` value is the same number used in the `mstconfig` command earlier in the procedure. 

. Enable secure boot
Enabling secure boot improves your system's security by preventing unauthorized operating systems and malicious software from loading during the device's boot process. 

.. Enable secure boot using the BIOS.
+
[source,yaml]
----
Secure Boot: Enabled
Secure Boot Policy: Standard
Secure Boot Mode: Mode Deployed
----

> .. Reboot the machine.

Author: Thanks @xenolinux for your feedback. I have expanded BIOS, added [source,terminal], and replaced machine with system.

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-sriov-device.adoc

[id="nw-sriov-nic-mlx-secure-boot_{context}"]
= MLX Secure Boot

> The Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Author: Updated with the ocpdocs-vale-bot recommendation.

> The Mellanox (MLX) secure boot supports the option to disable vendor plugins in the SR-IOV operator. This allows OCP customers to use the existing virtual functions in the mellanox cards firmware, and is currently the only way to use MLX NICs when secure boot is enabled. Secure boot is required by regulation and desired by customers.

Author: Updated with the ocpdocs-vale-bot recommendation.
